tests/test_add_model.py

```diff
@@ -0,0 +1,423 @@
+    mock_dir.return_value = ["resource_fields"]
+    mock_isinstance.return_value = False
+    mock_deduce_swagger_type.return_value = "dummy_swagger_type"
+
+    swagger.add_model(mock_model_class)
+
+    mock_dir.assert_called_with(mock_model_class)
+    assert mock_dir.call_count == 2
+    mock_hasattr.assert_called_once_with(mock_model_class, "required")
+    mock_isinstance.assert_called_with(mock_model_class, mock_nested)
+    assert mock_deduce_swagger_type.call_count == len(
+        mock_model_class.resource_fields.items()
+    )
+
+
+# Setup test for:
+#     * resource_fields: YES
+#     * nested subclass: YES
+#     * __init__: NO
+#     * swagger_metadata:NO
+
+test_fixtures_model_inputs = [
+    TodoItemWithResourceFields,
+]
+
+
+@patch("flask_restful_swagger.swagger.registry")
+@patch("flask_restful_swagger.swagger.deduce_swagger_type")
+@patch("flask_restful_swagger.swagger.isinstance")
+@patch("flask_restful_swagger.swagger.hasattr")
+@patch("flask_restful_swagger.swagger.dir")
+@patch("flask_restful_swagger.swagger._parse_doc")
+@pytest.mark.parametrize("model_class", test_fixtures_model_inputs)
+def test_add_model_with_resource_fields_with_nested(
+    mock_parse_doc,
+    mock_dir,
+    mock_hasattr,
+    mock_isinstance,
+    mock_deduce_swagger_type,
+    mock_registry,
+    model_class,
+):
+    _mock_registry = {"models": {}}
+    mock_registry.__getitem__.side_effect = _mock_registry.__getitem__
+    mock_registry.__setitem__.side_effect = _mock_registry.__setitem__
+    mock_parse_doc.return_value = (None, None)
+    mock_hasattr.return_value = True
+    mock_dir.return_value = ["resource_fields"]
+    mock_isinstance.return_value = True
+    mock_deduce_swagger_type.return_value = "dummy_swagger_type"
+
+    swagger.add_model(model_class)
+
+    mock_dir.assert_called_with(model_class)
+    assert mock_dir.call_count == 2
+    mock_hasattr.assert_called_once_with(model_class, "required")
+    mock_isinstance.assert_called_with(model_class, swagger._Nested)
+    assert mock_deduce_swagger_type.call_count == len(
+        model_class.resource_fields.items()
+    )
+
+
+# Setup test for:
+#     * resource_fields: YES
+#     * nested subclass: YES
+#     * __init__: NO
+#     * swagger_metadata:YES
+
+test_fixtures_model_inputs = [
+    ModelWithResourceFieldsWithRequiredWithSwaggerMetadata,
+]
+
+
+@patch("flask_restful_swagger.swagger.registry")
+@patch("flask_restful_swagger.swagger.deduce_swagger_type")
+@patch("flask_restful_swagger.swagger.isinstance")
+@patch("flask_restful_swagger.swagger.hasattr")
+@patch("flask_restful_swagger.swagger.dir")
+@patch("flask_restful_swagger.swagger._parse_doc")
+@pytest.mark.parametrize("model_class", test_fixtures_model_inputs)
+def test_add_model_with_resource_fields_nested_swagger_metadata(
+    mock_parse_doc,
+    mock_dir,
+    mock_hasattr,
+    mock_isinstance,
+    mock_deduce_swagger_type,
+    mock_registry,
+    model_class,
+):
+    _mock_registry = {"models": {}}
```

Comment:
Repeated.

Reply:
Refactored the data structure setup into seperate file `tests/fixtures_add_model.py`.  Using context managers to centralise the setup.